Step interpreter demo: Child blocks scatter when set block disconnected from parent

### Problem statement

Fireworks! Blocks are exploding! :)

- [x] Bug report
- [ ] Feature request


### Expected Behavior

![a9330987-f1a8-4739-a2c5-e31fdbf5cf07](https://user-images.githubusercontent.com/15675877/32809568-1f5bb2ec-c94c-11e7-8f76-18f34a2c4870.gif)

### Steps to Reproduce

1. Open up Step Interpreter demo.
2. Disconnect "set n" block from its parent.  
3. Watch them scatter! 

### Stack Traces

blockly_compressed.js:1297 Error: <g> attribute transform: Expected '(', "translate[object Object]".
Blockly.BlockSvg.setParent @ blockly_compressed.js:1297

### Operating System and Browser

_Remove this hint: these checkboxes can be checked like this: [x]_

- [ x] Desktop:
  - [x ] Chrome
  - [ ] Firefox
  - [ ] Safari
  - [ ] Opera
  - [ ] IE 10+
  - [ ] IE 11
  - [ ] EDGE

- [ ] Smartphone/Tablet/Chromebook (please complete the following information):
  - Device: [e.g. iPhone6]
  - OS: [e.g. iOS8.1]
  - Browser [e.g. stock browser, safari]
  - Version [e.g. 22]

### Additional Information

_Anything else we should know?_

--- a/core/block_svg.js
+++ b/core/block_svg.js
@@ -252,7 +252,7 @@ Blockly.BlockSvg.prototype.setParent = function(newParent) {
     // Move this block up the DOM.  Keep track of x/y translations.
     var xy = this.getRelativeToSurfaceXY();
     this.workspace.getCanvas().appendChild(svgRoot);
-    svgRoot.setAttribute('transform', 'translate' + xy);
+    svgRoot.setAttribute('transform', 'translate(' + xy.x + ',' + xy.y + ')');
   }
 
   Blockly.Field.startCache();
@@ -383,7 +383,7 @@ Blockly.BlockSvg.prototype.moveDuringDrag = function(newLoc) {
   if (this.useDragSurface_) {
     this.workspace.blockDragSurface_.translateSurface(newLoc.x, newLoc.y);
   } else {
-    this.svgGroup_.translate_ = 'translate' + newLoc;
+    this.svgGroup_.translate_ = 'translate(' + newLoc.x + ',' + newLoc.y + ')';
     this.svgGroup_.setAttribute('transform',
         this.svgGroup_.translate_ + this.svgGroup_.skew_);
   }

--- a/core/flyout_button.js
+++ b/core/flyout_button.js
@@ -189,7 +189,8 @@ Blockly.FlyoutButton.prototype.show = function() {
  * @private
  */
 Blockly.FlyoutButton.prototype.updateTransform_ = function() {
-  this.svgGroup_.setAttribute('transform', 'translate' + this.position_);
+  this.svgGroup_.setAttribute('transform',
+      'translate(' + this.position_.x + ',' + this.position_.y + ')');
 };
 
 /**

--- a/core/workspace_svg.js
+++ b/core/workspace_svg.js
@@ -705,7 +705,8 @@ Blockly.WorkspaceSvg.prototype.resetDragSurface = function() {
 
   var trans = this.workspaceDragSurface_.getSurfaceTranslation();
   this.workspaceDragSurface_.clearAndHide(this.svgGroup_);
-  var translation = 'translate' + trans + ' scale(' + this.scale + ')';
+  var translation = 'translate(' + trans.x + ',' + trans.y + ') ' +
+      'scale(' + this.scale + ')';
   this.svgBlockCanvas_.setAttribute('transform', translation);
   this.svgBubbleCanvas_.setAttribute('transform', translation);
 };

